tests/test_validators.py

```diff
@@ -1404,3 +1467,120 @@ def set_area(cls, values):
     assert r.area == 1
     r.height = 5
     assert r.area == 5
+
+
+V1_VALIDATOR_DEPRECATION_MATCH = (
+    r'Validator signatures using the `values` keyword argument or `\*\*kwargs` are no longer supported'
+)
+
+
+def _get_source_line(filename: str, lineno: int) -> str:
+    with open(filename) as f:
+        for _ in range(lineno - 1):
+            f.readline()
+        return f.readline()
+
+
+def _check_warning(warnings: List[WarningMessage]) -> None:
+    assert len(warnings) == 1
+    w = warnings[0]
+    # check that we got stacklevel correct
+    # if this fails you need to edit the stacklevel
+    # parameter to warnings.warn in _decorators.py
+    assert w.filename == __file__
+    source = _get_source_line(w.filename, w.lineno)
+    assert 'class Point(BaseModel):' in source
+
+
+def test_v1_validator_values_cls_method():
+    with pytest.warns(DeprecationWarning, match=V1_VALIDATOR_DEPRECATION_MATCH) as w:
+
+        class Point(BaseModel):
+            y: int
+            x: int
+
+            @validator('x')
+            def check_x(cls, y: Any, values: Dict[str, Any]) -> int:
+                raise NotImplementedError
+
+    _check_warning(w.list)
+
+
+def test_v1_validator_values_kwargs_cls_method():
+    with pytest.warns(DeprecationWarning, match=V1_VALIDATOR_DEPRECATION_MATCH) as w:
+
+        class Point(BaseModel):
+            y: int
+            x: int
+
+            @validator('x')
+            def check_x(cls, y: Any, values: Dict[str, Any], **kwargs: Any) -> int:
+                raise NotImplementedError
+
+    _check_warning(w.list)
+
+
+def test_v1_validator_kwargs_cls_method():
+    with pytest.warns(DeprecationWarning, match=V1_VALIDATOR_DEPRECATION_MATCH) as w:
+
+        class Point(BaseModel):
+            y: int
+            x: int
+
+            @validator('x')
+            def check_x(cls, y: Any, **kwargs: Any) -> int:
+                raise NotImplementedError
+
+    _check_warning(w.list)
+
+
+def test_v1_validator_values_function():
+    def _check_x(y: Any, values: Dict[str, Any]) -> int:
+        raise NotImplementedError
+
+    with pytest.warns(DeprecationWarning, match=V1_VALIDATOR_DEPRECATION_MATCH) as w:
+
+        class Point(BaseModel):
+            y: int
+            x: int
+
+            check_x = validator('x')(_check_x)
+
+    _check_warning(w.list)
+
+
+def test_v1_validator_values_kwargs_function():
+    def _check_x(y: Any, values: Dict[str, Any], **kwargs: Any) -> int:
+        raise NotImplementedError
+
+    with pytest.warns(DeprecationWarning, match=V1_VALIDATOR_DEPRECATION_MATCH) as w:
+
+        class Point(BaseModel):
+            y: int
+            x: int
+
+            check_x = validator('x')(_check_x)
```

Comment:
Do you have tests that these deprecated validator signatures work properly? It looks like this is just testing that the warning is emitted, but not that the validator actually works as intended. Seems to me to be worth adding tests that the v1 validator behavior still works with the deprecated signature, at least as long as we intend for it to. (And I'm assuming we do for the purpose of migration-friendliness.)

Reply:
https://github.com/pydantic/pydantic/blob/73088c94368f7ce29e779f640b892c563b774b88/tests/test_validators.py#L1495-L1586

Reply:
These tests aren't checking that the validator is actually called though, right? Am I missing something? I meant something like:
```python
def _check_x(v):
    return v * 2

assert Point(x=1, y=1).model_dump() == {'x': 2, 'y': 1}
```
(Mostly just making sure it actually calls the validator during validation without erroring.)

If you think it's unnecessary I'm fine, just felt like it was possible that these tests could pass and yet the actual validation cause errors for other reasons.

Reply:
No it’s a good idea! I’ll add them

Reply:
Donzo, thank you for the good idea!